requirements.txt

```diff
@@ -1,9 +1,24 @@
-ruamel.yaml == 0.17.10
-titlecase == 2.3
-requests == 2.26.0
-beautifulsoup4 == 4.9.3
-lxml == 4.6.3
-patool == 1.12
-
-# PyLogger submodule
-colorama == 0.4.4
\ No newline at end of file
+attrs==21.2.0
+beautifulsoup4==4.9.3
+certifi==2021.5.30
+charset-normalizer==2.0.4
+colorama==0.4.4
+idna==3.2
+iniconfig==1.1.1
+lxml==4.6.3
+marshmallow==3.13.0
+mokkari==0.1.10
+packaging==21.0
+patool==1.12
+pluggy==0.13.1
+py==1.10.0
+pyparsing==2.4.7
+pytest==6.2.4
+ratelimit==2.2.1
+requests==2.26.0
+ruamel.yaml==0.17.10
+ruamel.yaml.clib==0.2.6
+soupsieve==2.2.1
+titlecase==2.3
+toml==0.10.2
+urllib3==1.26.6
```

Comment:
I don't think its necessary to include all these dependencies, testing locally should only need the below
```suggestion
ruamel.yaml == 0.17.10
titlecase == 2.3
requests == 2.26.0
beautifulsoup4 == 4.9.3
lxml == 4.6.3
patool == 1.12
colorama == 0.4.4
mokkari == 0.1.10

# For Testing
pytest == 6.2.4
```

Reply:
I'm fairly certain they are needed for the various dependencies. Should be easy enough to check just do a 'pip3 install mokkari pytest', then 'pip3 freeze > requirements.txt'.

Reply:
`pip freeze` takes a snapshot of all the libraries in the pip env not just the ones you install. I like to have just the install dependencies in the requirements.txt instead of the dependencies of dependencies as they get installed anyway.

Reply:
Depends on what your goals for the project is, if it's just an internal project for yourself that fine, but if it's not, you're going to want to have reproducible builds to save yourself future headaches. Also, if your primary concern is having the Pipfile list only the primary deps, you're better off using one of the next-generation dependency management tools like pipenv or poetry.

Reply:
Awesome, I thought I was doing it the standard way, I'll merge this in and start looking into Poetry and see how that might fit into this project. thanks

Reply:
Haven't used poetry myself, but I've heard good things.